From 95f16aaf75eea437702e359075613fd80d562049 Mon Sep 17 00:00:00 2001
From: David Redondo <qt@david-redondo.de>
Date: Wed, 7 Oct 2026 20:56:05 +0200
Subject: [PATCH] QDesktopUnixServices: Send multiple email recipients
 correctly to the portal

The path of a mailto URI can contain multiple addresses delimited by ",".
Additionally a "to" field can be present. To send multiple addresses to
the portal we have to use the "addresses" field.

Change-Id: I0014b8315eb785552340903239784a48908c846d
Reviewed-by: Liang Qi <liang.qi@qt.io>
(cherry picked from commit c0cdef12ae11526e98b07b1148a7b33fc9012ff9)
Reviewed-by: David Redondo <qt@david-redondo.de>
Origin: upstream, https://invent.kde.org/qt/qt/qtbase/-/commit/95f16aaf75eea437702e359075613fd80d562049
Forwarded: not-needed
Last-Update: 2026-10-29
---
 src/gui/platform/unix/qdesktopunixservices.cpp | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/src/gui/platform/unix/qdesktopunixservices.cpp b/src/gui/platform/unix/qdesktopunixservices.cpp
index 6a0e9ef5e4b..1883cd295aa 100644
--- a/src/gui/platform/unix/qdesktopunixservices.cpp
+++ b/src/gui/platform/unix/qdesktopunixservices.cpp
@@ -260,8 +260,9 @@ static inline QDBusMessage xdgDesktopPortalSendEmail(const QUrl &url, const QStr
 
     QUrlQuery urlQuery(url);
     QVariantMap options;
-    if (const auto address = url.path(); !address.isEmpty())
-        options.insert("address"_L1, address);
+    QStringList addresses = url.path().split(u',') +  urlQuery.allQueryItemValues("to"_L1, QUrl::FullyDecoded);
+    if (!addresses.isEmpty())
+        options.insert("addresses"_L1, addresses);
     if (const auto subject = urlQuery.queryItemValue("subject"_L1); !subject.isEmpty())
         options.insert("subject"_L1, subject);
     if (const auto body = urlQuery.queryItemValue("body"_L1); !body.isEmpty())
